research-software-directory/RSD-as-a-service
# 1343 category checkboxes

# Add checkboxes to all levels of organsation and community categories

Closes #1343 
Closes #1348

Changes proposed in this pull request:
* Altered data generation module for creating catgories. Generate category names have "global, organisation or community" word to indicate the type of category.
* When deleting organisation from software/project related categories are also deleted.
* Add checkboxes to all levels of categories for organisation software/projects. Selecting "lower" level item results is automatical selection of "higer" levels. Deselecting "higer" level item results in automatic "deselection" of all children items.
* Add option to "fold" child items in order to shorten the list length (for easier reading)
* This functionality is implemented only for organisation and community related categories. **Global categories, used only for software are not included in this PR**.

How to test:
* `make start` to build and generate test data
* login as rsd admin
* select software with organisation categories. Try to change values at different levels and see what your experience is with new approach. This is POC to discuss.
* select project with organisation categories and test the categories approach. The working should be the same as for software.
* enable communities module in the settings.json and test the functionality for community categories
* add also global categories to software. Confirm that global categories have checkboxes only at "lowest" level.
* confirm that categories are ordered alphabetically by name

**Remark**
Further improvements might be needed concerning the order on the software and project page. For software, in particular, we can have 3 types of categories: global, organisation and community specific categories. Currently all category groups/types are listed in alphabetical order.

![image](https://github.com/user-attachments/assets/b19c592a-986a-46b0-a183-241afada005f)

## Example global categories (not changed)
![image](https://github.com/user-attachments/assets/25e402c7-da0a-46ed-8414-f4a206a5da93)

## Example organisation/community categories (changed)
![image](https://github.com/user-attachments/assets/4276312a-739c-4480-ad10-c09dad516664)

PR Checklist:
